proofs/tree_test.go

```diff
@@ -1316,6 +1316,38 @@ func TestCreateProof_standard_customEncoder(t *testing.T) {
 	assert.EqualError(t, err, "Hash does not match")
 }
 
+func TestCreateOneofProof(t *testing.T) {
+	doctree := NewDocumentTree(TreeOptions{Hash: sha256Hash})
+	err := doctree.AddLeavesFromDocument(&documentspb.ExampleOneofSampleDocument, &documentspb.ExampleOneofSampleSaltsDocument)
+	assert.Nil(t, err)
+	err = doctree.Generate()
+
+	_, err = doctree.CreateProof("InexistentField")
+	assert.EqualError(t, err, "No such field: InexistentField in obj")
+
+	proof, err := doctree.CreateProof("valueB")
+	assert.Nil(t, err)
+	assert.Equal(t, ReadableName("valueB"), proof.Property)
```

Comment:
Could you create a test for the following additional scenario:  Try to create a proof for valueC when valueB is set

Reply:
And when both are nil create a proof for either one (or even both) when both are nil.